src/praisonai-agents/praisonaiagents/agent/tool_execution_fixed.py

```
@@ -0,0 +1,132 @@
+            
+            for param_name, arg_value in arguments.items():
+                if param_name in sig.parameters:
+                    param = sig.parameters[param_name]
+                    if param.annotation != inspect.Parameter.empty:
+                        # Try to cast to the expected type
+                        annotation = param.annotation
+                        # Handle Optional types
+                        if hasattr(annotation, '__origin__'):
+                            if annotation.__origin__ is Union:
+                                # Get non-None type from Union
+                                types = [t for t in annotation.__args__ if t != type(None)]
```

Ruff (0.15.18)

[error] Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)